src/commands/capacity.ts

```diff
@@ -0,0 +1,48 @@
+import { CommandDefinition } from "../Command";
+import { TournamentStatus } from "../database/interface";
+import { ChallongeTournament } from "../database/orm";
+import { reply } from "../util/discord";
+import { UserError } from "../util/errors";
+import { getLogger } from "../util/logger";
+
+const logger = getLogger("command:capacity");
+
+const command: CommandDefinition = {
+	name: "capacity",
+	requiredArgs: ["id"],
+	executor: async (msg, args, support) => {
+		const [id, rawCapacity] = args; // 1 optional and thus potentially undefined
+		const capacity = parseInt(rawCapacity, 10);
+		logger.verbose(
+			JSON.stringify({
+				channel: msg.channel.id,
+				message: msg.id,
+				user: msg.author.id,
+				tournament: id,
+				command: "capacity",
+				capacity
+			})
+		);
+		if (rawCapacity && (isNaN(capacity) || capacity < 0 || `${capacity}` !== rawCapacity)) {
```

Comment:
This is a cool check on parseInt's results that I hadn't considered before

Reply:
I used something like it in the YDK parser only to have to revert it when certain inputs had leading zeros